File: Ridy-taxi-app/android/rider/src/main/java/com/innomalist/taxi/rider/networking/socket/CalculateFare.kt
package com.innomalist.taxi.rider.networking.socket

import com.google.android.gms.maps.model.LatLng
import com.innomalist.taxi.common.models.ServiceCategory
import com.innomalist.taxi.common.networking.socket.interfaces.SocketRequest
import com.squareup.moshi.JsonClass
import org.json.JSONArray
import org.json.JSONObject

class CalculateFare(locations: List<LatLng>) : SocketRequest() {
    init {
        val arr = JSONArray()
        for(loc in locations) {
            val obj = JSONObject()
            obj.put("x", loc.longitude)
            obj.put("y", loc.latitude)
            arr.put(obj)
        }
        this.params = arrayOf(arr)
    }
}

@JsonClass(generateAdapter = true)
data class CalculateFareResult(
        val categories: List<ServiceCategory>,
        val distance: Int,
        val duration: Int,
        val currency: String
)

File: Ridy-taxi-app/ios/rider/rider/Networking/WebSocket/CalculateFare.swift
//
//  CalculateFare.swift
//  rider
//
//  Created by Manly Man on 11/26/19.
//  Copyright © 2019 minimal. All rights reserved.
//

import UIKit
import MapKit

class CalculateFare: SocketRequest {
    typealias ResponseType = CalculateFareResult
    var params: [Any]?
    
    init(locations: [CLLocationCoordinate2D]) {
        self.params = [locations.map() { loc in
            return [
                "x": loc.longitude,
                "y": loc.latitude
            ]
        }]
    }

}

struct CalculateFareResult: Codable {
    var categories: [ServiceCategory]
    var distance: Int
    var duration: Int
    var currency: String
}
